src/base/static/components/templates/main-map.js

```diff
@@ -246,84 +400,101 @@ class MainMap extends Component {
-        latitude={this.props.mapViewport.latitude}
-        longitude={this.props.mapViewport.longitude}
-        pitch={this.props.mapViewport.pitch}
-        bearing={this.props.mapViewport.bearing}
-        zoom={this.props.mapViewport.zoom}
-        transitionDuration={this.props.mapViewport.transitionDuration}
-        transitionInterpolator={this.props.mapViewport.transitionInterpolator}
-        transitionEasing={this.props.mapViewport.transitionEasing}
-        mapboxApiAccessToken={MAP_PROVIDER_TOKEN}
-        minZoom={this.props.mapViewport.minZoom}
-        maxZoom={this.props.mapViewport.maxZoom}
-        onMouseUp={this.endFeatureQuery}
-        onMouseDown={this.beginFeatureQuery}
-        onTouchEnd={this.onMouseUp}
-        onTouchStart={this.onMouseDown}
-        onViewportChange={viewport => {
-          // NOTE: react-map-gl seems to cache the width and height of the map
-          // container at the beginning of a transition. If the viewport change
-          // that initiated the transition also changed the width and/or height
-          // (for example when a Place is clicked and the map resizes to make
-          // way for the content panel), then react-map-gl will immediately
-          // undo the width and height change. The result of this is yet
-          // another version of the off-center bug.
-          //
-          // So, we strip out the height and width from any viewport changes
-          // originating from react-map-gl. There should never be a situation
-          // where react-map-gl needs to set the width or height of the map
-          // container.
-          const { width, height, ...rest } = viewport;
-          this.props.updateMapViewport(
-            rest,
-            this.isMapTransitioning
-              ? false
-              : this.props.mapConfig.options.scrollZoomAroundCenter,
-          );
-        }}
-        onTransitionStart={() => (this.isMapTransitioning = true)}
-        onTransitionEnd={() => (this.isMapTransitioning = false)}
-        interactiveLayerIds={this.props.interactiveLayerIds}
-        mapStyle={this.props.mapStyle}
-        onInteractionStateChange={this.onInteractionStateChange}
-        onLoad={this.onMapLoad}
-      >
-        {this.state.isMapLoaded && (
-          <MapControlsContainer>
-            <NavigationControl
-              onViewportChange={viewport =>
-                this.props.updateMapViewport(viewport)
-              }
-            />
-            <GeolocateControl
-              updateMapViewport={this.props.updateMapViewport}
-            />
-            {this.props.leftSidebarConfig.panels.map(panel => (
-              <CustomControl
-                key={panel.id}
-                icon={panel.icon}
-                component={panel.component}
-                setLeftSidebarExpanded={this.props.setLeftSidebarExpanded}
-                setLeftSidebarComponent={this.props.setLeftSidebarComponent}
+      <Fragment>
+        <Global
```

Comment:
This is a hack to make `mapbox-gl-draw` work with `react-map-gl`. Basically we need to clear `react-map-gl`'s overlay responsible for capturing mouse events out of the way so clicks will register with `mapbox-gl-draw`'s geometry.

Reply:
This is an emotion 10.x dependency. We are still on emotion 9: https://github.com/jalMogo/mgmt/issues/242

Instead of having emotion 10 and 9 bundled into our app, I suggest exploring ways to inject global styles using our current emotion 9 lib: https://www.npmjs.com/package/emotion#global-styles

But if that's not possible, this approach seems ok for now :+1:

Reply:
Thanks for pointing this out-- I didn't realize I was mixing major versions :/.

I might be missing a better way, but I don't think the the 9.x global facilities will work here, since we need to be able to declaratively set the `pointerEvents` property on a global selector whenever the value of the `isDrawModeActive` prop.

I tried this approach with 9.x's `injectGlobal` used in the `render()` method of `MainMap`:

```
injectGlobal`
  .overlays {
    pointer-events: ${this.props.isDrawModeActive ? "none" : "initial"}
  }
`;
```

But it doesn't seem to update the `pointer-events` property in the same way. Maybe it injects a new global selector each time it's run?

Reply:
Sounds tricky, and I might not have time to get into it. I'm fine with mixing major versions of Emotion for now.

Reply:
Ok, I'll merge this as-is then and prioritize upgrading Emotion project-wide.